tests/core/test_context.py

```diff
@@ -274,3 +293,39 @@ def demo(named: Named) -> None:
     spec = empty_context._type_spec_for_function(demo)
     assert spec["named"][0] is hasname_ref
     assert spec["named"][1].default_factory is None
+
+
+def test_str_default(
+    empty_context: BlueskyContext, sim_motor: SynAxis, alt_motor: SynAxis
+):
+    movable_ref = empty_context._reference(Movable)
+    empty_context.device(sim_motor)
+    empty_context.plan(has_default_reference)
+
+    spec = empty_context._type_spec_for_function(has_default_reference)
+    assert spec["m"][0] is movable_ref
+    assert spec["m"][1].default_factory() is SIM_MOTOR_NAME
```

Comment:
`==` instead of `is`? Does the instance matter?

Reply:
Done